src/core/grid.hpp

```diff
@@ -91,44 +63,30 @@ extern Utils::Vector3d my_right;
     determine one automatically. */
 void init_node_grid();
 
-/** node mapping: array -> node.
- *
- * \param node   rank of the node you want to know the position for.
- * \param pos    position of the node in node grid.
- */
-void map_node_array(int node, int pos[3]);
-
-/** node mapping: node -> array.
- *
- * \return      rank of the node at position pos.
- * \param pos   position of the node in node grid.
- */
-int map_array_node(Utils::Span<const int> pos);
-
 /** map a spatial position to the node grid */
 int map_position_node_array(const Utils::Vector3d &pos);
 
 /** fill neighbor lists of node.
  *
- * Calculates the numbers of the nearest neighbors for a node and
- * stores them in \ref node_neighbors.
+ * Calculates the numbers of the nearest neighbors for a node.
  *
- * \return     the number of neighbors
- * \param node number of the node.  */
-int calc_node_neighbors(int node);
+ * \return Ranks of neighbors
+ */
+Utils::Vector<int, 6> calc_node_neighbors(const boost::mpi::communicator &comm);
+
+/**
+ * @brief Calculate the position of node in topology.
+ *
+ * @param comm Cartesian communicator
+ * @return Index of node in grid.
+ */
+Utils::Vector3i calc_node_pos(const boost::mpi::communicator &comm);
```

Comment:
why don't we use https://www.boost.org/doc/libs/1_68_0/doc/html/boost/mpi/cartesian_communicator.html
?

Reply:
Because that is not available in all boost versions that we support. There is a price for everything...